tests/vmi_networking_test.go

```diff
@@ -691,27 +691,44 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 	})
 
 	Context("VirtualMachineInstance with masquerade binding mechanism", func() {
+		const (
+			defaultCIDR = false
+			customCIDR  = true
+		)
+
 		var serverVMI *v1.VirtualMachineInstance
 		var clientVMI *v1.VirtualMachineInstance
 
-		masqueradeVMI := func(containerImage string, userData string, Ports []v1.Port) *v1.VirtualMachineInstance {
+		masqueradeVMI := func(Ports []v1.Port, ipv4NetworkCIDR string) *v1.VirtualMachineInstance {
+			containerImage := tests.ContainerDiskFor(tests.ContainerDiskCirros)
+			userData := "#!/bin/bash\necho 'hello'\n"
 			vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(containerImage, userData)
 			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{{Name: "default", Ports: Ports, InterfaceBindingMethod: v1.InterfaceBindingMethod{Masquerade: &v1.InterfaceMasquerade{}}}}
-			vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+			net := v1.DefaultPodNetwork()
+			if ipv4NetworkCIDR != "" {
+				net.NetworkSource.Pod.VMNetworkCIDR = ipv4NetworkCIDR
+			}
+			vmi.Spec.Networks = []v1.Network{*net}
 
 			return vmi
 		}
 
-		table.DescribeTable("[test_id:1780]should allow regular network connection", func(ports []v1.Port) {
+		table.DescribeTable("[test_id:1780]should allow regular network connection", func(ports []v1.Port, withCustomCIDR bool) {
+			var ipv4NetworkCIDR string
+
+			if withCustomCIDR {
+				ipv4NetworkCIDR = "10.10.10.0/24"
+			}
```

Comment:
wouldn't it be better if you injected this from the caller ? e.g. instead of a `withCustomCIDR bool` , have a `customCIDR *string ` ?

Reply:
The test needs to check custom CIDR, so this flag is here for both IPv4 and IPv6.
It looked nicer to me to read the test this way, over passing in empty strings on the other two scenarios.

As there is no intention to test now permutations between CIDR with or without IPv6, not different addresses, this seemed better.

Reply:
> The test needs to check custom CIDR, so this flag is here for both IPv4 and IPv6.

Hm, this might change my opinion; could you paste a snippet of how you envision the test body with IPv6 ? Just some high level pseudo-code is enough.

> It looked nicer to me to read the test this way, over passing in empty strings on the other two scenarios.

You don't have to pass empty strings around; define `customCIDR` as a pointer, and use nil.

> 
> As there is no intention to test now permutations between CIDR with or without IPv6, not different addresses, this seemed better.

Reply:
```go
table.FDescribeTable("[test_id:1780]should allow regular network connection", func(ports []v1.Port, withCustomCIDR bool) {
        var ipv4NetworkCIDR string
        var ipv6NetworkCIDR string

        if withCustomCIDR {
                ipv4NetworkCIDR = "10.10.10.0/24"
                ipv6NetworkCIDR = "fd10:10:10::/120"
        }
```

Reply:
I'm convinced.